base/reducedim.jl

```diff
@@ -1070,7 +1074,57 @@ function findminmax!(f, op, Rval, Rind, A::AbstractArray{T,N}) where {T,N}
                     Rval[i,IR] = tmpAv
                     Rind[i,IR] = k
                 end
-                y = iterate(ks, kss)
+                y = iterate(keys, kss)
+            end
+        end
+    end
+    Rval, Rind
+end
+
+# Optimized fallback for `keys` with the same axes, e.g. `LinearIndices`, `CartesianIndices`.
+# We initialize `Rval`/`Rind` via `map/copyfirst!` to support non-1 based `A`.
+function _findminmax!(f, op, Rval, Rind, A::AbstractArray{T,N}, keys::AbstractArray{<:Any,N}, init = true) where {T,N}
+    axes(keys) == axes(A) || return @invoke _findminmax!(f, op, Rval, Rind, A::AbstractArray, keys::Any, init)
+    (isempty(Rval) || isempty(A)) && return Rval, Rind
+    lsiz = check_reducedims(Rval, A)
+    for i = 1:N
+        axes(Rval, i) == axes(Rind, i) || throw(DimensionMismatch("Find-reduction: outputs must have the same indices"))
+    end
+    if init
+        copyfirst!(Rind, keys)
+        mapfirst!(f, Rval, A)
```

Comment:
Nice catch. Before this PR, the only usage for `mapfirst!` are the initialization for `maximum!`.
The redundancy also happens there as `Base._mapreducedim!` also performs `check_reducedims`.
I think it's ok to remove the `check_reducedims` within `mapfirst!` as it use `view` rather than `@inbounds view`.